src/app.js

```diff
@@ -452,18 +452,32 @@ var run = function () {
 
   var previousInput = ''
   var compileTimeout = null
+  var saveTimeout = null
 
   function editorOnChange () {
     var input = editor.getValue()
-    if (input === '') {
-      editor.setCacheFileContent('')
-      return
-    }
+
+    // if there's no change, don't do anything
     if (input === previousInput) {
       return
     }
     previousInput = input
 
+    // fire storage update
+    // NOTE: save at most once per 5 seconds
+    if (saveTimeout) {
+      window.clearTimeout(saveTimeout)
+    }
+    saveTimeout = window.saveTimeout(function () {
```

Comment:
I see, cause i never saw `window.saveTimeout`
it should be `window.setTimeout` I think
And also we should retrigger the timer each time. setTimeout does not act as a loop.. but maybe this is already done elsewhere.